plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -6255,18 +6255,23 @@ public void testReadFromVersionedTableWithExpiredHistory()
     }
 
     @Test
-    public void testDeleteRetainsTableHistory()
+    public void testDeleteRetainsTableHistoryAndMetadataFile()
     {
         String tableName = "test_delete_retains_table_history_" + randomNameSuffix();
         assertUpdate("CREATE TABLE " + tableName + "(c1 INT, c2 INT)");
         assertUpdate("INSERT INTO " + tableName + " VALUES (1, 1), (2, 2), (3, 3)", 3);
         assertUpdate("INSERT INTO " + tableName + " VALUES (3, 3), (4, 4), (5, 5)", 3);
         List<Long> snapshots = getTableHistory(tableName);
+        List<Long> metadataLogEntries = getTableMetadataLogEntries(tableName);
 
         assertUpdate("DELETE FROM " + tableName + " WHERE c1 < 4", 4);
         List<Long> snapshotsAfterDelete = getTableHistory(tableName);
         assertThat(snapshotsAfterDelete.size()).isGreaterThan(snapshots.size());
         assertThat(snapshotsAfterDelete).containsAll(snapshots);
+
+        List<Long> metadataLogEntriesAfterDelete = getTableMetadataLogEntries(tableName);
```

Comment:
The changes in this test seem unrelated to the current commit.
Pls add them in a separate commit.